src/main/java/org/opensearch/knn/index/memory/NativeMemoryCacheManager.java

```diff
@@ -68,20 +68,27 @@ public static synchronized NativeMemoryCacheManager getInstance() {
     }
 
     private void initialize() {
+        initialize(
+            KNNSettings.state().getSettingValue(KNNSettings.KNN_MEMORY_CIRCUIT_BREAKER_ENABLED),
+            KNNSettings.getCircuitBreakerLimit().getKb(),
+            KNNSettings.state().getSettingValue(KNNSettings.KNN_CACHE_ITEM_EXPIRY_ENABLED),
+            ((TimeValue) KNNSettings.state().getSettingValue(KNNSettings.KNN_CACHE_ITEM_EXPIRY_TIME_MINUTES)).getMinutes()
+        );
+    }
+
+    private void initialize(boolean isWeightLimited, long maxWeight, boolean isExpirationLimited, long expiryTimeInMin) {
```

Comment:
nit: I suggest we use DTO instead of 4 arguments. It's will increase maintainability for future changes if we need to add/remove settings, especially as we can re-use DTO in multiple places, e.g. rebuildCache, tests etc.

Reply:
Sure, will update